controllers/envtest/suite_test.go

```diff
@@ -0,0 +1,234 @@
+
+	volumegroupv1 "github.com/IBM/csi-volume-group-operator/api/v1"
+	"github.com/IBM/csi-volume-group-operator/controllers"
+	"github.com/IBM/csi-volume-group-operator/controllers/envtest/utils"
+	"github.com/IBM/csi-volume-group-operator/controllers/persistentvolumeclaim"
+	"github.com/IBM/csi-volume-group-operator/controllers/volumegroupcontent"
+	"github.com/IBM/csi-volume-group-operator/pkg/client/fake"
+	"github.com/IBM/csi-volume-group-operator/pkg/config"
+	"github.com/IBM/csi-volume-group-operator/tests/mock_grpc_server"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	cancel    context.CancelFunc
+	ctx       context.Context
+	server    *mock_grpc_server.MockServer
+)
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Controller Suite")
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	var err error
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	err = volumegroupv1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	//+kubebuilder:scaffold:scheme
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	server, err = mock_grpc_server.CreateMockServer()
+	Expect(err).ToNot(HaveOccurred())
+	addr := server.Address()
+	csiConn, err := fake.New(addr, DriverName)
+	Expect(err).ToNot(HaveOccurred())
+	driverConfig := &config.DriverConfig{
+		DriverName:        DriverName,
+		DriverEndpoint:    addr,
+		RPCTimeout:        time.Minute,
+		MultipleVGsToPVC:  "false",
+		DisableDeletePvcs: "false",
+	}
+	mockVolumeGroup := fake.VolumeGroup{
+		CreateVolumeGroupMock: func(name string, secrets, parameters map[string]string) (*csi.CreateVolumeGroupResponse, error) {
+			return &csi.CreateVolumeGroupResponse{}, nil
+		},
+		DeleteVolumeGroupMock: func(volumeGroupId string, secrets map[string]string) (*csi.DeleteVolumeGroupResponse, error) {
+			return &csi.DeleteVolumeGroupResponse{}, nil
+		},
+		ModifyVolumeGroupMembershipMock: func(volumeGroupId string, volumeIds []string, secrets map[string]string) (*csi.ModifyVolumeGroupMembershipResponse, error) {
+			return &csi.ModifyVolumeGroupMembershipResponse{}, nil
+		},
+	}
+
+	err = (&controllers.VolumeGroupReconciler{
+		Client:       mgr.GetClient(),
+		Scheme:       mgr.GetScheme(),
+		DriverConfig: driverConfig,
+		Log:          ctrl.Log.WithName("controllers").WithName("VolumeGroup"),
```

Comment:
any reason why its not like other reconcilers
`Log:          ctrl.Log.WithName("VolumeGroupController"),`

Reply:
I did it as we do in the [`main.go`](https://github.com/IBM/csi-volume-group-operator/blob/develop/main.go#L92-L98) file

Reply:
ok